plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/AuthorizationCodeReceiver.java

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.google.cloud.tools.eclipse.appengine.login;
+
+import com.google.common.base.Preconditions;
+
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Runs a Jetty web server instance that opens up an HTTP listening port to receive OAuth2
+ * authorization code.
+ */
```

Comment:
Talked to @nbashirbello and found out that [google-oauth-java-client](https://github.com/google/google-oauth-java-client) already has the [`LocalServerReceiver`](https://github.com/google/google-oauth-java-client/blob/dev/google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java) implementation. I spent some time to use it, but eventually I realized it has a critical problem: you are blocked on `waitForCode()` indefinitely until the server receives login result. I will not go into the details, but this must have been one of the reasons that our IntelliJ plugin duplicated the `LocalServerReceiver` code. It makes sense to do the same thing in Eclipse too, so I'll continue with the current state of `AuthorizationCodeReceiver`.

As a side note, `VerificationCodeReceiver` is YAGNI.